code/fireball/fireballs.cpp

```diff
@@ -999,6 +1031,8 @@ float fireball_wormhole_intensity(fireball *fb)
 	return rad;
 } 
 
+extern void warpin_queue_render(model_draw_list *scene, object *obj, matrix *orient, vec3d *pos, int texture_bitmap_num, float radius, float life_percent, float max_radius, bool warp_3d, int warp_glow_bitmap, int warp_ball_bitmap, int warp_model_id);
```

Comment:
Why did you declare this function here and not in the header?

Reply:
It was previously in the header; I removed the header declaration and added this.  I was thinking that header files are ideally suited for common functions used in many places, and specifically intended as interfaces.  Since this function was only used in warpineffect.cpp in one location, there was no harm in the extern.

And yes, this isn't consistent across the project, but it was the same thinking as extern'ing `fireball_parse_tbl` in the other PR.

Reply:
I'm generally not in favor for `extern` declarations since they are prone for errors and they make the code less modular but it doesn't really matter here.